lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1561,7 +1556,13 @@ protected getCachedTokenInternal(scopes : Array<string> , user: User): CacheResu
   
     try {
         if (tokenReceivedCallback) {
-            tokenReceivedCallback.call(self, errorDesc, token, error, tokenType,  this.getUserState(this._cacheStorage.getItem(Constants.stateLogin, this.storeAuthStateInCookie)));
+            //We should only send the stae back to the developer if it matches with what we received from the server
+            if (requestInfo.stateMatch) {
+                tokenReceivedCallback.call(self, errorDesc, token, error, tokenType, this.getUserState(requestInfo.stateResponse));
```

Comment:
we are missing the this.storeAuthStateInCookie

Reply:
As discussed, we don't need that flag

Reply:
As discussed, we don't require this flag